core/src/local/persistent_state.rs

```diff
@@ -382,19 +370,19 @@ impl PersistentState {
     //     Uniquely identifies the index in self.indices.
     //  * `row_size`
     //      The byte size of `row` when serialized with binode.
-    //  * `row`
+    //  * `key`
     //      The actual index values.
     //  * `epoch`
     //      Incremented on each PersistentState initialization.
     //  * `seq`
     //      Sequence number since last epoch.
-    fn serialize_key(&self, index_id: IndexID, row: &Vec<&DataType>, seq: IndexSeq) -> Vec<u8> {
-        let size: u64 = bincode::serialized_size(&row).unwrap();
-        bincode::serialize(&(index_id, size, row, self.epoch, seq)).unwrap()
+    fn serialize_key(&self, index_id: IndexID, key: &KeyType, seq: IndexSeq) -> Vec<u8> {
```

Comment:
Hmm.. I think we'd rather serialize the keys the way we used to (i.e., without the `KeyType` enum variant field). It'd basically mean matching on `KeyType` here and then serializing the elements individually. Don't know how important saving 8 bytes on every key is though..